### DISCUSSION TITLE
How to Fix Remarks at Page Bottom and Reset Page Numbers per Order in BIRT 4.9.0?

### DISCUSSION THREAD

**Discussion by asdfgh282812** [eclipse-birt/birt#1962] [Q&A]
Hello everyone,

I'm currently using BIRT 4.9.0.v201905231911 to design a picking list report that needs to be exported in PDF format. I've encountered some issues during the design process and would like to seek your help.

Requirements:

### **Remarks Position Issue:**

I need to display the "Remarks" information for each shipper (order) fixed at the bottom of each page, regardless of the number of data rows. The remarks should always appear at the very bottom of the page, not immediately after the data rows.
### Independent Page Numbering:

I want each order to start on a new page, with independent page numbering.
No matter how many orders are in a batch, each order should be treated as a separate page or page group, and the page number should display as "Page X of Y", where X is the current page number, and Y is the total number of pages for that order, not the entire report.

**What I've Tried:**

### Remarks Position Issue:

In the group footer, I added a grid and tried to set the placeholder row's height and the "Can Shrink" & "Can Grow" properties to fix the remarks at the bottom of the page.
Problem:
I couldn't find the "Can Shrink" & "Can Grow" options in BIRT 4.9.0 (this was suggested by ChatGPT).
It's difficult to precisely control the height of the placeholder row, and the remarks position still varies with the amount of data.

### Independent Page Numbering:

In the group properties, I tried to check the options "Insert Page Break at Start of Group" and "Reset Page Number" (this was suggested by ChatGPT).
Problem:
I'm not sure if these options exist in BIRT 4.9.0 because I couldn't find them.
Although the page number starts from 1, the total page number still reflects the total pages of the entire report, not the number of pages for each order.
My Environment:

BIRT Version: 4.9.0.v201905231911
Report Output Format: PDF
I Would Like to Ask:

Regarding the Remarks Position:

Is there a way in BIRT 4.9.0 to fix the group footer or other elements at the bottom of the page, so that the remarks always appear at the very bottom, unaffected by the number of data rows?
If there are no "Can Shrink" and "Can Grow" properties, is there another method to achieve this effect?
Regarding Independent Page Numbering:

Do the options "Insert Page Break at Start of Group" and "Reset Page Number" exist in BIRT 4.9.0's group properties?
If not, how can I set up the report so that each order starts on a new page, and the page numbering resets, showing the total pages only for the current order?
Thank you very much for your help! If you need more information or if anything is unclear, please let me know.

Here is the conversation content with Chat GPT (in Chinese, translation provided if needed)
https://chatgpt.com/share/672c6239-2510-8008-9de9-8c5301d82c20

**Comment by hvbtup** [comment]
Regarding "independent page numbering", search for "grouped page numbering". 
I guess it is possible with separate RunTask and RenderTask, but I don't know how.
I use a PDF post processor which adds the page numbers based on the PDF outline (an invisible TOC in a PDF file) instead. You can use the "TOC entry" field in the group editor dialog to create TOC entries.

Regarding the footer fixed at the bottom somehow, I don't think it is possible without patching BIRT. In my fork at https://github.com/triestram-partner/birt (branch tup_main) I did this, based on a User Property: 
https://github.com/eclipse-birt/birt/commit/cc8302352ca3e6d8ac09ee87866d58b3f5692fac
But I never actually used this in production.

The `canShrink` attribute is interesting. I really don't know its purpose. A `canGrow` attribute does not exist.

**Reply by asdfgh282812** [reply]
Thank you for your response! For the page numbering, I ultimately achieved it using a database approach, which allowed me to bypass BIRT entirely. As for the fixed footer, I used a variable mapping to handle it, though this method requires placing the content on the page and then hiding it. While not ideal, it did effectively resolve the issue.
